### PR TITLE
added time-dependent box size

### LINKED ISSUE #361: Reads incorrect (open) boundary conditions when using H5MD trajectory file
Hi,

I tried coarse-graining the atomistic trajectory. But votca uses open boundary conditions instead of periodic boundary conditions.  
I run the command:
```
csg_map --top topology.xml --trj AAtrajectory.h5 --out output.dump --cg mapping.xml
The output is:
I have 2010 beads in 134 molecules
I have 134 beads in 134 molecules for the coarsegraining
H5MD: has /box/edges
H5MD: has /force
Reading frame: 0NOTE: You are using OpenBox boundary conditions. Check if this is intended.
```
Please find the files attached.
[votca_issue.zip](https://github.com/votca/csg/files/3034324/votca_issue.zip)

Thanks,
Aditi.

### PR DESCRIPTION
**Pull request by junghans**
Fix #361 
Fix #362

### CODE REVIEW

**Comment by junghans** [comment]
@AditiKhot please test this patch!

File: src/libcsg/modules/io/h5mdtrajectoryreader.cc
Comment (JoshuaSBrown) [inline src/libcsg/modules/io/h5mdtrajectoryreader.cc]: Is there anyway we can avoid using new and raw pointers here?

Comment (JoshuaSBrown) [reply src/libcsg/modules/io/h5mdtrajectoryreader.cc]: maybe declare unique_ptr and pass data_out.get() to H5Dread, then you don't need to worry about deleting it later.

**Review by JoshuaSBrown — CHANGES_REQUESTED** [review]
Also address the Codacy issue raised.

**Comment by junghans** [comment]
Can you take a shot at it?

**Comment by JoshuaSBrown** [comment]
So, we do not yet have support for make_unique in std c++11 I think, hence the error. You will have to use unique_ptr<>() instead of make_unique<>().

**Comment by junghans** [comment]
@JoshuaSBrown can you do a pass?

**Comment by JoshuaSBrown** [comment]
Ok, but you will owe me a cider, I'm not a big beer drinker.

**Comment by JensWehner** [comment]
I thought we are on c++14 now?

**Comment by JensWehner** [comment]
ahh no stable is still c++11

**Review by JensWehner — APPROVED** [review]
if it is c++11 compliant it looks good

**Review by JoshuaSBrown — APPROVED** [review]

**Review by jkrajniak — APPROVED** [review]

**Comment by AditiKhot** [comment]
> @AditiKhot please test this patch!

I get this error when I recompile votca:
namespace "std" has no member "make_unique"

**Comment by AditiKhot** [comment]
> > @AditiKhot please test this patch!
> 
> I get this error when I recompile votca:
> namespace "std" has no member "make_unique"

I think I was using wrong files. However, I still get an error as below while compiling

`In file included from /depot/bsavoie/apps/VOTCA/votca/csg/include/votca/csg/beadlist.h(23),
                 from /depot/bsavoie/apps/VOTCA/votca/csg/src/libcsg/beadlist.cc(18):
/depot/bsavoie/apps/VOTCA/votca/csg/include/votca/csg/topology.h(434): error: no instance of constructor "votca::csg::Bead::Bead" matches the argument list
            argument types are: (votca::csg::Topology *, std::size_t, std::string, votca::tools::byte_t, std::string, int, double, double)
        Bead *b = new Bead(this, _beads.size(), type, symmetry, name, resnr, m, q);
                          ^
/depot/bsavoie/apps/VOTCA/votca/csg/include/votca/csg/bead.h(44): note: this candidate was rejected because mismatch in count of arguments
  class Bead : public TopologyItem
        ^
/depot/bsavoie/apps/VOTCA/votca/csg/include/votca/csg/bead.h(386): note: this candidate was rejected because arguments do not match
      Bead(Topology *owner, int id, BeadType *type, byte_t symmetry, string name, int resnr, double m, double q)
      ^

compilation aborted for /depot/bsavoie/apps/VOTCA/votca/csg/src/libcsg/beadlist.cc (code 2)
`
Thank you for the help!

**Comment by junghans** [comment]
@AditiKhot don't worry the patch already got merged in `stable`, so can do https://github.com/votca/votca/blob/master/share/doc/INSTALL.md#general-source-installation-instructions with `version=stable`, or wait until #364 is merged and you can use `master` as well.

**Comment by AditiKhot** [comment]
I tried using stable version, but I get the above error.

**Pull request merged**
